test/NuGet.Core.Tests/NuGet.Protocol.Core.v3.Tests/LocalFolderUtilityTests.cs

```diff
@@ -14,6 +14,337 @@ namespace NuGet.Protocol.Core.v3.Tests
+            }
+        }
+
+        [Fact]
+        public void LocalFolderUtility_GetPackagesConfigFolderPackages_ById()
+        {
+            using (var root = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                // Arrange
+                var testLogger = new TestLogger();
+                var a = new PackageIdentity("a", NuGetVersion.Parse("1.0.0"));
+                var a2 = new PackageIdentity("a", NuGetVersion.Parse("1.0.0-beta"));
+                var b = new PackageIdentity("b", NuGetVersion.Parse("1.0.0"));
+                var c = new PackageIdentity("c", NuGetVersion.Parse("1.0.0"));
+
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, a);
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, a2);
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, b);
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, c);
+
+                // Act
+                var packages = LocalFolderUtility.GetPackagesConfigFolderPackages(root, "a", testLogger)
+                    .OrderBy(package => package.Identity.Id)
+                    .ThenBy(package => package.Identity.Version)
+                    .ToList();
+
+                // Assert
+                Assert.Equal(2, packages.Count);
+                Assert.Equal("a.1.0.0-beta", packages[0].Identity.ToString());
+                Assert.Equal("a.1.0.0", packages[1].Identity.ToString());
+            }
+        }
+
+        [Fact]
+        public void LocalFolderUtility_GetPackagesConfigFolderPackages_ById_NotFound()
+        {
+            using (var root = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                // Arrange
+                var testLogger = new TestLogger();
+                var a = new PackageIdentity("a", NuGetVersion.Parse("1.0.0"));
+                var a2 = new PackageIdentity("a", NuGetVersion.Parse("1.0.0-beta"));
+                var b = new PackageIdentity("b", NuGetVersion.Parse("1.0.0"));
+                var c = new PackageIdentity("c", NuGetVersion.Parse("1.0.0"));
+
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, a);
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, a2);
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, b);
+                SimpleTestPackageUtility.CreateFolderFeedPackagesConfig(root, c);
+
+                // Act
+                var packages = LocalFolderUtility.GetPackagesConfigFolderPackages(root, "z", testLogger)
+                    .OrderBy(package => package.Identity.Id)
+                    .ThenBy(package => package.Identity.Version)
+                    .ToList();
+
+                // Assert
+                Assert.Equal(0, packages.Count);
+            }
+        }
+
+        [Fact]
+        public void LocalFolderUtility_GetPackagesConfigFolderPackage()
```

Comment:
while going through the code, i saw you have fallback code to check through all the sub-folders in case you don't find the exact match... so I'm thinking there will be cases where it falls in that code path?